src/views/CountryRiskWatch/HistoricalDataChart/index.tsx

```diff
@@ -0,0 +1,446 @@
+import {
+    Fragment,
+    useMemo,
+    useRef,
+} from 'react';
+import {
+    isDefined,
+    isFalsyString,
+    isNotDefined,
+    unique,
+} from '@togglecorp/fujs';
+import {
+    DroughtIcon,
+    FloodIcon,
+    FoodSecurityIcon,
+    StormIcon,
+} from '@ifrc-go/icons';
+
+import { useRequest } from '#utils/restRequest';
+import type { paths } from '#generated/types';
+import Container from '#components/Container';
+import SelectInput from '#components/SelectInput';
+import DropdownMenu from '#components/DropdownMenu';
+import TextOutput from '#components/TextOutput';
+import useTranslation from '#hooks/useTranslation';
+import useInputState from '#hooks/useInputState';
+import useSizeTracking from '#hooks/useSizeTracking';
+import { numericIdSelector, stringNameSelector } from '#utils/selectors';
+import { getPrettyBreakableBounds, getScaleFunction } from '#utils/chart';
+import { formatNumber, sumSafe } from '#utils/common';
+import {
+    COLOR_HAZARD_CYCLONE,
+    COLOR_HAZARD_DROUGHT,
+    COLOR_HAZARD_FLOOD,
+    COLOR_HAZARD_FOOD_INSECURITY,
+} from '#utils/constants';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type GetGoHistorical = paths['/api/v2/go-historical/']['get'];
+type GoHistoricalQuery = GetGoHistorical['parameters']['query'];
+type GoHistoricalResponse = GetGoHistorical['responses']['200']['content']['application/json'];
+type EventItem = NonNullable<GoHistoricalResponse['results']>[number];
+type PartialDisasterType = EventItem['dtype'];
+type DisasterType = Omit<PartialDisasterType, 'name'> & {
+    name: string;
+}
+
+const DISASTER_FLOOD = 12;
+const DISASTER_FLASH_FLOOD = 27;
+const DISASTER_CYCLONE = 4;
+const DISASTER_FOOD_INSECURITY = 21;
+const DISASTER_DROUGHT = 20;
```

Comment:
Any reason we are hard-coding these?

Reply:
Yes, we only need those hazards

Reply:
But why are we hardcoding these? How do we know that `20` is always going to be DISASTER_DROUGHT

Reply:
Yess!

Reply:
I'll add a TODO to add flag in the database for the future work